### PR TITLE
Remove symbols files for final 3 libraries

### PR DESCRIPTION
**Pull request by sbc100**
Just link these by default (unless AUTO_LIBRARIES=0 is used).

### CODE REVIEW

File: src/settings.js [hunk 1]
```diff
@@ -1583,6 +1584,11 @@ var SINGLE_FILE = 0;
 // library function in library_foo.js.
 var AUTO_JS_LIBRARIES = 1;
 
+// Like AUTO_JS_LIBRARIES but for the natice libraries such as libgl, libal
```
Comment (kripken) [inline src/settings.js:1587]:
```suggestion
// Like AUTO_JS_LIBRARIES but for the native libraries such as libgl, libal
```

File: src/settings.js [hunk 2]
```diff
@@ -1583,6 +1584,11 @@ var SINGLE_FILE = 0;
...
+// and libhtml5.   If this is disabled it is necessary to explcitly add
+// -lhtml5 and additional to have pre-built the library using `embuilder`.
```
Comment (kripken) [inline src/settings.js:1589]:
```suggestion
// e.g. -lhtml5 and also to first build the library using `embuilder`.
```

File: src/settings.js [hunk 3]
```diff
@@ -1583,6 +1584,11 @@ var SINGLE_FILE = 0;
...
+var AUTO_LIBRARIES = 1;
```
Comment (kripken) [inline src/settings.js:1590]: maybe `AUTO_NATIVE_LIBRARIES`? otherwise it is not parallel to `AUTO_JS_LIBRARIES`

**Comment by sbc100** [comment]
> Another option instead of adding a new setting here is to require these be added with `-lhtml5` etc. That might be too much of a breaking change though, especially for html5.
> 
> Worth verifying this has no impact on link times.

This is exactly that `-s STRICT` mode forces users to do.  I say we land this change than look a making AUTO_LIBRARIES default to 0 as a followup.

**Comment by kripken** [comment]
sgtm

**Comment by sbc100** [comment]
Trying once again to land this.  Added a ChangeLog entry this time.

File: ChangeLog.md [hunk 4]
```diff
@@ -20,6 +20,12 @@ See docs/process.md for more on how version tagging works.
 
 Current Trunk
 -------------
+- The system for linking native libraries on demand based on the symbols
+  present in input object files has been removed.  Libraries such as libgl,
+  libal, and libhtml5 are now linked in by default unless
+  `-s AUTO_NATIVE_LIBRARIES=0` is used.  This should not effect most builds
+  since wasm-ld will only link the objects withing these libraries if they
```
Comment (kripken) [inline ChangeLog.md:27]:
```suggestion
  since wasm-ld will only link the objects within these libraries if they
```

**Review by kripken — APPROVED** [review]
lgtm with confirmation that there is no effect on link times.

Why mention in the changelog if this is not noticeable? In case there is a bug?

**Comment by sbc100** [comment]
I shouldn't have any affect on link times unless you are using `--whole-archive` .. which is what MAIN_MODULE=1 does.  This means that MAIN_MODULE=1 programs will be slightly larger then before but they are already HUGE and the indented to include all libraries.. and we have an option to directly this feature if somebody cares enough about their MAIN_MODULE=1 build

**Comment by sbc100** [comment]
I couldn't measure any impact on the link time of hello world (i used the a super small program becuase those are the ones where one might possibly see the effect of adding unused libraries to the link line).

**Pull request merged**